tla/consistency/MultiNode.tla

```diff
@@ -0,0 +1,54 @@
+---- MODULE MultiNode ----
+\* This specification extends SingleNode to model a multi-node CCF service
+
+EXTENDS SingleNode
+
+\* Upper bound on the view
+CONSTANT ViewLimit
```

Comment:
How about refactoring `ViewLimit` and `HistoryLimit` in `SingleNode` into a TLC state constraint.  Especially, because the term "view" also appears in `raft.h`.

Reply:
I can add a state constraint using `TLCGet("level")` but it would be useful to keep the option to set views and history length separately. I can write a state constraint using views and history length but I'm not sure how to express the idea that once view limit is reach, the model can continue but one action (which increases view) cannot be taken

Reply:
Try `SomeActionConstraint == TLCGet("level") > 42 => ~OneAction`

Reply:
Sorry, I've not used action constraints before. I'm struggling to specify them in the cfg file, I've tried `ACTION-CONSTRAINT SomeActionConstraint` and  `ACTION-CONSTRAINTS SomeActionConstraint` to no avail. See screenshots below 

![image](https://github.com/microsoft/CCF/assets/1835251/31e316b4-6c7e-4182-8526-ebf7044a009f)

Reply:
I've solved it, I needed to use `ACTION_CONSTRAINT` instead of `ACTION-CONSTRAINT`, based on the comment here https://github.com/tlaplus/tlaplus/blob/a41cbafc66b1dd225156aaca38ad35ec330f4ae9/toolbox/org.lamport.tla.toolbox.tool.tlc/src/org/lamport/tla/toolbox/tool/tlc/launch/TLCModelLaunchDelegate.java#L588

Reply:
I have put in a PR to update cfg file syntax hosted by Apalache https://github.com/informalsystems/apalache/pull/2752.

Reply:
Instead of an ACTION_CONSTRAINT, consider adding the following to the MC*.tla file:

```tla
SN == INSTANCE SingleNode

ACOriginalAction ==
    /\ TLCGet("level") < 7
    /\ SN!OriginalAction
```

MC*.cfg
```
CONSTANT
   OriginalAction <- ACOriginalAction
```

Compare e.g. https://github.com/microsoft/CCF/blob/e73cf2f4cfb9a6231f1d52c729bb60094a1c14b0/tla/MCccfraft.tla#L19-L29

Reply:
Sorry, I should have updated this thread. I've actually already done that, see the MC* files in PR currently

Reply:
Hint: The `INSTANCE-OpRedefinition` trick has the advantage that you don't have to redefine the spec formulas. :-)